pkg/prometheus/statefulset.go

```diff
@@ -565,311 +400,80 @@ func makeStatefulSetSpec(
-					Name: "OBJSTORE_CONFIG",
-					ValueFrom: &v1.EnvVarSource{
-						SecretKeyRef: p.Spec.Thanos.ObjectStorageConfig,
-					},
-				})
-			}
-			thanosArgs = append(thanosArgs, monitoringv1.Argument{Name: "tsdb.path", Value: storageDir})
-			container.VolumeMounts = append(
-				container.VolumeMounts,
-				v1.VolumeMount{
-					Name:      volName,
-					MountPath: storageDir,
-					SubPath:   subPathForStorage(p.Spec.Storage),
-				},
-			)
-
-			// NOTE(bwplotka): As described in https://thanos.io/components/sidecar.md/ we have to turn off compaction of Prometheus
-			// to avoid races during upload, if the uploads are configured.
-			disableCompaction = true
-		}
-
-		if p.Spec.Thanos.TracingConfig != nil || len(p.Spec.Thanos.TracingConfigFile) > 0 {
-			if len(p.Spec.Thanos.TracingConfigFile) > 0 {
-				thanosArgs = append(thanosArgs, monitoringv1.Argument{Name: "tracing.config-file", Value: p.Spec.Thanos.TracingConfigFile})
-			} else {
-				thanosArgs = append(thanosArgs, monitoringv1.Argument{Name: "tracing.config", Value: "$(TRACING_CONFIG)"})
-				container.Env = append(container.Env, v1.EnvVar{
-					Name: "TRACING_CONFIG",
-					ValueFrom: &v1.EnvVarSource{
-						SecretKeyRef: p.Spec.Thanos.TracingConfig,
-					},
-				})
-			}
-		}
-
-		if p.Spec.Thanos.LogLevel != "" {
-			thanosArgs = append(thanosArgs, monitoringv1.Argument{Name: "log.level", Value: p.Spec.Thanos.LogLevel})
-		} else if p.Spec.LogLevel != "" {
-			thanosArgs = append(thanosArgs, monitoringv1.Argument{Name: "log.level", Value: p.Spec.LogLevel})
-		}
-		if p.Spec.Thanos.LogFormat != "" {
-			thanosArgs = append(thanosArgs, monitoringv1.Argument{Name: "log.format", Value: p.Spec.Thanos.LogFormat})
-		} else if p.Spec.LogFormat != "" {
-			thanosArgs = append(thanosArgs, monitoringv1.Argument{Name: "log.format", Value: p.Spec.LogFormat})
-		}
-
-		if p.Spec.Thanos.MinTime != "" {
-			thanosArgs = append(thanosArgs, monitoringv1.Argument{Name: "min-time", Value: p.Spec.Thanos.MinTime})
-		}
-
-		if p.Spec.Thanos.ReadyTimeout != "" {
-			thanosArgs = append(thanosArgs, monitoringv1.Argument{Name: "prometheus.ready_timeout", Value: string(p.Spec.Thanos.ReadyTimeout)})
-		}
-
-		containerArgs, err := operator.BuildArgs(thanosArgs, p.Spec.Thanos.AdditionalArgs)
-		if err != nil {
-			return nil, err
-		}
-		container.Args = append([]string{"sidecar"}, containerArgs...)
-
-		additionalContainers = append(additionalContainers, container)
-	}
-	if disableCompaction {
-		thanosBlockDuration := operator.StringValOrDefault(string(p.Spec.Thanos.BlockDuration), "2h")
-		promArgs = append(promArgs, monitoringv1.Argument{Name: "storage.tsdb.max-block-duration", Value: thanosBlockDuration})
-		promArgs = append(promArgs, monitoringv1.Argument{Name: "storage.tsdb.min-block-duration", Value: thanosBlockDuration})
-	}
+	if disableCompaction {
+		thanosBlockDuration := operator.StringValOrDefault(string(thanos.BlockDuration), "2h")
```

Comment:
there's a potential nil pointer exception which I didn't catch in #5360 :)
```suggestion
		thanosBlockDuration := "2h"
		if thanos != nil {
			thanosBlockDuration := string(thanos.BlockDuration)
		}
```

Reply:
good catch